From 0fe5bfc65bcb6213b88bd17bab7ce4ae61c366ba Mon Sep 17 00:00:00 2001
From: Mohan Prabu <quic_mprabub@quicinc.com>
Date: Mon, 2 Sep 2024 16:57:28 +0530
Subject: [PATCH] cfg80211: extend remove links for STA

Since links are controlled by userspace via cfg80211
in STA mode also, remove them from the driver in that
case.

Change-Id: I64448d5ca9def423b987ebfafa3a8785777e4cd2
Signed-off-by: Mohan Prabu <quic_mprabub@quicinc.com>
---
 net/wireless/util.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/net/wireless/util.c b/net/wireless/util.c
index a1f018cf4fd6..8702b0624cb0 100644
--- a/net/wireless/util.c
+++ b/net/wireless/util.c
@@ -2672,9 +2672,10 @@ void cfg80211_remove_links(struct wireless_dev *wdev)
 
 	/*
 	 * links are controlled by upper layers (userspace/cfg)
-	 * only for AP mode, so only remove them here for AP
+	 * only for AP and STA mode, so only remove them here for AP or STA
 	 */
-	if (wdev->iftype != NL80211_IFTYPE_AP)
+	if (wdev->iftype != NL80211_IFTYPE_AP &&
+		wdev->iftype != NL80211_IFTYPE_STATION)
 		return;
 
 	wdev_lock(wdev);
-- 
2.34.1

